exo/networking/manual/manual_discovery.py

```diff
@@ -62,8 +75,35 @@ async def task_find_peers_from_config(self):
             try: del self.known_peers[peer_id]
             except KeyError: pass
         except Exception as e:
-            if DEBUG_DISCOVERY >= 2: print(f"Exception occured when attempting to add {peer_id=}: {e}")
+          if DEBUG_DISCOVERY >= 2: print(f"Exception occured when attempting to add {peer_id=}: {e}")
       await asyncio.sleep(1.0)
 
       if DEBUG_DISCOVERY >= 2: print(f"Current known peers: {[peer.id() for peer in self.known_peers.values()]}")
 
+  def _get_peers(self):
+    try:
+      current_mtime = os.path.getmtime(self.network_config_path)
+
+      if self._cached_peers is not None and self._last_modified_time is not None and current_mtime <= self._last_modified_time:
+        return self._cached_peers
+
+      topology = NetworkTopology.from_path(self.network_config_path)
+
+      if self.node_id not in topology.peers:
+        raise ValueError(
+          f"Node ID {self.node_id} not found in network config file "
+          f"{self.network_config_path}. Please run with `node_id` set to "
+          f"one of the keys in the config file: {[k for k, _ in topology.peers]}"
+        )
+
+      peers_in_network: Dict[str, PeerConfig] = topology.peers
+      peers_in_network.pop(self.node_id)
+
+      self._cached_peers = peers_in_network
+      self._last_modified_time = current_mtime
+
+      return peers_in_network
+
+    except Exception as e:
+      if DEBUG_DISCOVERY >= 2: print(f"Error when loading network config file from {self.network_config_path}. Please update the config file in order to successfully discover peers. Exception: {e}")
+      return self._cached_peers
```

Comment:
The only gotcha I found with handling dynamic updates to the config file is if someone saves a WIP/invalid json (instead of pasting in a fully new & correct json blob).

So the current solution is to just swallow any exception we get when reading from the config and return the cached config (until the config file is set correctly)

Reply:
That's a fine solution.